api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/controller/AccountsController.java

```diff
@@ -217,40 +208,48 @@ public ResponseEntity<Object> deletePlugin(@ApiParam(value = "provide account nu
     @ApiOperation(httpMethod = "POST", value = "API to create plugin", response = Response.class, produces = MediaType.APPLICATION_JSON_VALUE)
     @PostMapping(path = "/{type}/create", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<Object> createPlugin(@AuthenticationPrincipal Principal user,
-                                               @RequestBody RedHatPluginRequest request,
+                                               @RequestBody Object request,
                                                @PathVariable("type") String type) {
+        PluginResponse response;
         try {
             String createdBy = amazonCognitoConnector.getCognitoUserDetails(user.getName()).getEmail();
-            PluginResponse response = redHatPluginService.createPlugin(request, createdBy);
-            if (response.getStatus().equals(AdminConstants.SUCCESS)) {
-                List<String> optionalAssetList = null;
-                if (Objects.nonNull(optionalAssetGroupList)) {
-                    optionalAssetList = Arrays.asList(optionalAssetGroupList.split(","));
-                }
-                if (Objects.nonNull(optionalAssetList) && !optionalAssetList.isEmpty()
-                        && optionalAssetList.contains(type)) {
-                    String updateAssetGroupStatus = assetGroupService.updateAssetGroupStatus(type, true, "admin");
-                    log.info("AssetGroup  {} status {}", type, updateAssetGroupStatus);
-                }
-                log.info(type + " plugin created successfully.");
+            PluginParameters parameters = PluginParameters.builder().pluginName(type).createdBy(createdBy).build();
+            if (type.equals("redhat")) {
```

Comment:
I am assuming you are simply reusing RedHat and GCP create plugin endpoint, anyway we can make it more generic?

For example, if the UI sends you a map of params to add to secret manager, we should be able to do it generically and the secret path will be dictated by tenant/datasource/account.

Thoughts and LOE please. 

PS: we can also do another cleanup task for this.